sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -1165,11 +1165,14 @@ pi_result cuda_piContextRelease(pi_context ctxt) {
     CUcontext cuCtxt = ctxt->get();
     CUcontext current = nullptr;
     cuCtxGetCurrent(&current);
-    if(cuCtxt != current)
-    {
-      PI_CHECK_ERROR(cuCtxSetCurrent(cuCtxt));
+    if (cuCtxt != current) {
```

Comment:
Please, explain what the problem is in the commits message and add a test.

Reply:
Description has been added to both the commit and this PR. The bugs that this fixes are some odd corner cases and are difficult to test.

Reply:
Give me a moment. I pushed something to the wrong branch.